src/ngRoute/route.js

```diff
@@ -160,7 +171,7 @@ function $RouteProvider() {
    */
   this.when = function(path, route) {
     //copy original route object to preserve params inherited from proto chain
-    var routeCopy = angular.copy(route);
+    var routeCopy = createShallowCopy(route);
```

Comment:
ok. sgtm.

On Thu, Jun 9, 2016 at 2:25 PM Georgios Kalpakas <notifications@github.com>
wrote:

> In src/ngRoute/route.js
> <https://github.com/angular/angular.js/pull/14750#discussion_r66524807>:
>
> > @@ -160,7 +171,7 @@ function $RouteProvider() {
> >     */
> >    this.when = function(path, route) {
> >      //copy original route object to preserve params inherited from proto chain
> > -    var routeCopy = angular.copy(route);
> > +    var routeCopy = createShallowCopy(route);
>
> Since we are in a external module, we need to copy it. I started by
> copying that, but then removed the stuff that didn't seem relevant.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/angular/angular.js/pull/14750/files/f4b882c8c30d0d0d68fe04ed65bfca1640cfe0cf#r66524807>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AANM6M5AG48ivz-eHgiQTGEgpn15Qbnpks5qKITJgaJpZM4IyaSo>
> .
>